src/main/java/Nell.java

```diff
@@ -0,0 +1,217 @@
+        System.out.println("-> The task has been added to the list:");
+        ToDo toDoToAdd = new ToDo(commandBody);
+        listAddTask(toDoToAdd);
+        System.out.println("   " + toDoToAdd);
+        System.out.println(String.format("   The list now has %d tasks", taskCount));
+    }
+
+    /**
+     * Executes a deadline command with a given command body
+     *
+     * @param commandBody The command body
+     */
+    private static void executeCommandDeadline(String commandBody) {
+        System.out.println("-> The task has been added to the list:");
+        String[] commandWords = commandBody.split("/by");
+        Deadline deadlineToAdd = new Deadline(commandWords[0].trim(), commandWords[1].trim());
+        listAddTask(deadlineToAdd);
+        System.out.println("   " + deadlineToAdd);
+        System.out.println(String.format("   The list now has %d tasks", taskCount));
+    }
+
+    /**
+     * Executes an event command with a given command body
+     *
+     * @param commandBody The command body
+     */
+    private static void executeCommandEvent(String commandBody) {
+        System.out.println("-> The task has been added to the list:");
+        String[] commandWords = commandBody.split("/from|/to", 3);
+        Event eventToAdd = new Event(commandWords[0].trim(), commandWords[1].trim(), commandWords[2].trim());
+        listAddTask(eventToAdd);
+        System.out.println("   " + eventToAdd);
+        System.out.println(String.format("   The list now has %d tasks", taskCount));
+    }
+
+    /**
+     * Executes an unmark command with a given command body
+     *
+     * @param commandBody The command body
+     */
+    private static void executeCommandUnmark(String commandBody) {
+        int taskIndex = Integer.parseInt(commandBody);
+        if (checkIfTaskValid(taskIndex)) {
+            // Marks a task as not done
+            tasks[taskIndex - 1].setDone(false);
+            System.out.println("-> The following task has been marked not done:");
+            printTaskAtIndex(tasks[taskIndex - 1], taskIndex);
+        } else {
+            System.out.println("-> Invalid task!");
+        }
+    }
+
+    /**
+     * Executes a mark command with a given command body
+     *
+     * @param commandBody The command body
+     */
+    private static void executeCommandMark(String commandBody) {
+        int taskIndex = Integer.parseInt(commandBody);
+        if (checkIfTaskValid(taskIndex)) {
+            // Marks a task as done
+            tasks[taskIndex - 1].setDone(true);
+            System.out.println("-> The following task has been marked done:");
+            printTaskAtIndex(tasks[taskIndex - 1], taskIndex);
+        } else {
+            System.out.println("-> Invalid task!");
+        }
+    }
+
+    /**
+     * Executes an add task command (no command word) with a given command body
+     *
+     * @param command The command body
+     */
+    private static void executeCommandAddTask(String command) {
```

Comment:
Instead of including the execute functions in the Nell class, might want to consider creating a separate class to include these functionalities like (CommandManager or something) to delegate command management and improve abstraction in your code this way

Reply:
I agree with this, thanks for your suggestion!